ext/openssl/ossl_x509store.c

```diff
@@ -329,7 +328,23 @@ ossl_x509store_set_trust(VALUE self, VALUE trust)
 static VALUE
 ossl_x509store_set_time(VALUE self, VALUE time)
 {
-    rb_iv_set(self, "@time", time);
+    X509_STORE *store;
+    X509_VERIFY_PARAM *param;
+
+    GetX509Store(self, store);
+#if HAVE_X509_STORE_GET0_PARAM
+    param = X509_STORE_get0_param(store);
+#else
+    param = X509_VERIFY_PARAM_new();
```

Comment:
Struct fields on `X509_STORE` (and in general) were part of public API in OpenSSL 1.0.2.

```suggestion
    param = store->param;
```

The `X509_STORE_set1_param()` call later will become unnecessary with this.